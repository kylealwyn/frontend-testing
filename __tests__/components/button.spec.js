import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Button from '../../src/components/button';

describe('<Button />', () => {
  let wrapper;
  let spy;
  const text = 'Hello Jest!';

  beforeEach(() => {
    spy = jest.fn();

    wrapper = shallow(
      <Button onClick={spy}>
        {text}
      </Button>
    );
  });

  afterEach(() => {
    spy.mockReset();
  });

  it('should render a button', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct children text', () => {
    expect(wrapper.text()).toEqual(text);
  });

  it('should call the `onClick` prop when clicked', () => {
    expect(spy.mock.calls.length).toEqual(0);
    wrapper.find('.btn').simulate('click');
    expect(spy.mock.calls.length).toEqual(1);
  })

  it('should test instance methods', () => {
    const favoriteColor = wrapper.instance().getFavoriteColor();
    expect(favoriteColor).toBe('blue');
  });
});
