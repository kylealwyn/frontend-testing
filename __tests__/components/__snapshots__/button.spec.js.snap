// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button 1`] = `
<button
  className="btn"
  onClick={[Function]}
>
  Hello Jest!
</button>
`;
